Scripts/XRLineRenderer.cs

```diff
@@ -0,0 +1,399 @@
+﻿using UnityEngine;
+using UnityEngine.Serialization;
+
+/// <summary>
+/// An XR-Focused drop-in replacement for the Line Renderer
+/// This renderer draws fixed-width lines with simulated volume and glow.
+/// This has many of the advantages of the traditional Line Renderer, old-school system-level line rendering functions, 
+/// and volumetric (a linked series of capsules or cubes) rendering
+/// </summary>
+[RequireComponent(typeof(MeshRenderer))]
+[RequireComponent(typeof(MeshFilter))]
+[ExecuteInEditMode]
+public class XRLineRenderer : XRLineRendererBase
+{
+      // Stored Line Data
+    [SerializeField]
+    [Tooltip("All of the connected points to render as a line.")]
+    Vector3[] m_Positions;
+
+    [SerializeField]
+    [FormerlySerializedAs("m_WorldSpaceData")]
+    [Tooltip("Draw lines in worldspace (or local space) - driven via shader.")]
+    bool m_UseWorldSpace;
+
+    [SerializeField]
+    [Tooltip("Connect the first and last vertices, to create a loop.")]
+    bool m_Loop;
+    
+    /// <summary>
+    /// Draw lines in worldspace (or local space)
+    /// </summary>
+    public bool useWorldSpace
+    {
+        get
+        {
+            return m_UseWorldSpace;
+        }
+        set
+        {
+            m_UseWorldSpace = value;
+        }
+    }
+
+    /// <summary>
+    /// Makes sure that the internal world space flag of the line renderer
+    /// matches the world space flag of the first material on the object
+    /// </summary>
+    void CopyWorldSpaceDataFromMaterial()
+    {
+        if (m_Materials != null && m_Materials.Length > 0)
+        {
+            var firstMaterial = m_Materials[0];
+            if (firstMaterial == null)
+            {
+                return;
+            }
+            if (firstMaterial.HasProperty("_WorldData"))
+            {
+                m_UseWorldSpace = !Mathf.Approximately(firstMaterial.GetFloat("_WorldData"), 0.0f);
+            }
+            else
+            {
+                m_UseWorldSpace = false;
+            }
```

Comment:
According to our standards doc for this one particular case (brackets on one-line ifs) you can use them or not, either way is fine